web/src/lib/components/login/Nostr.svelte

```diff
@@ -0,0 +1,87 @@
+<script lang="ts">
+    import { createEventDispatcher, onMount } from 'svelte';
+    import { ErrorHandler, loginNostr } from "$lib/services/api";
+    import type { User } from "$lib/types/user";
+    import { token, user, Info } from "$lib/stores";
+
+    const dispatch = createEventDispatcher();
+
+    export let onLogin: (user: User | null) => void = (_) => {};
+
+    enum Step {
+        SetKey,
+        SetVerificationPhrase,
+    }
+
+    let step: Step;
+
+    let npub: string | null = null;
+    let verificationPhrase: string | null = null;
+
+    function sendVerificationPhrase() {
+        if (npub === null) {
+            return;
+        }
+
+        loginNostr(npub, null,
+            () => {
+                Info.set("Sent the verification phrase!");
+                step = Step.SetVerificationPhrase;
+            });
+    }
+
+    let verifying = false;
+    function verify() {
+        if (npub === null || verificationPhrase === null) {
+            return;
+        }
+
+        verifying = true;
+        loginNostr(npub, verificationPhrase, () => verifying = false,
+            async (response) => {
+                verifying = false;
+                token.set(response.token);
+                localStorage.setItem('token', response.token);
+                dispatch('login', {})
+
+                while ($user === null) {
+                    await new Promise(resolve => setTimeout(resolve, 100));
+                }
+
+                onLogin(response.user);
+            },
+            new ErrorHandler(true, () => verifying = false));
+    }
+
+    onMount(() => {
+        step = Step.SetKey;
+    });
+</script>
+
+{#if step === Step.SetKey}
+    <div class="w-full flex items-center justify-center mt-4">
+        <div class="form-control w-full max-w-full">
+            <label class="label" for="npub">
+                <span class="label-text">Your NPUB</span>
+            </label>
+            <input bind:value={npub} type="text" name="npub" class="input input-lg input-bordered" />
```

Comment:
I know that you plan to improve the Nostr login using extensions in the future, without sending the words and everything, but in this screen, you could improve the UX x100 with just 3 lines of code by allowing the people that have a Nostr extension to auto-get the npub. In `nos2x` it's very easy to get it, but with `Alby`, I needed like 10 clicks to know where to get the npub from...

If you show a button `Get from extension` just when the user `hasExtension()`, and then when clicked:

```
let pubkey = await window.nostr.getPublicKey();
npub = nip19.npubEncode(pubkey);
```
it would be SOOOO much better... :)